plotly/tools.py

```diff
@@ -1451,6 +1455,823 @@ class FigureFactory(object):
+                         'Earth': ['rgb(0,0,130)', 'rgb(255,255,255)'],
+                         'Electric': ['rgb(0,0,0)', 'rgb(255,250,220)'],
+                         'Viridis': ['rgb(68,1,84)', 'rgb(253,231,37)']}
+
+        # Validate colors
+        if colors is None:
+            colors = DEFAULT_PLOTLY_COLORS
+
+        if isinstance(colors, str):
+            if colors in plotly_scales:
+                colors = plotly_scales[colors]
+
+            elif 'rgb' in colors:
+                # validate rgb color
+                colors = FigureFactory._unlabel_rgb(colors)
+                for value in colors:
+                    if value > 255.0:
+                        raise exceptions.PlotlyError("Whoops! The "
+                                                     "elements in your "
+                                                     "rgb colors "
+                                                     "tuples cannot "
+                                                     "exceed 255.0.")
+                colors = FigureFactory._label_rgb(colors)
+
+                # put colors in list
+                colors_list = []
+                colors_list.append(colors)
+                colors = colors_list
+
+            elif '#' in colors:
+                colors = FigureFactory._hex_to_rgb(colors)
+                colors = FigureFactory._label_rgb(colors)
+
+                # put colors in list
+                colors_list = []
+                colors_list.append(colors)
+                colors = colors_list
+
+            else:
+                scale_keys = list(plotly_scales.keys())
+                raise exceptions.PlotlyError("If you input a string "
+                                             "for 'colors', it must "
+                                             "either be a Plotly "
+                                             "colorscale, an 'rgb' "
+                                             "color or a hex color."
+                                             "Valid plotly colorscale "
+                                             "names are {}".format(scale_keys))
+        elif isinstance(colors, tuple):
+            # validate tuple color
+            for value in colors:
+                if value > 1.0:
+                    raise exceptions.PlotlyError("Whoops! The "
+                                                 "elements in "
+                                                 "your colors "
+                                                 "tuples cannot "
+                                                 "exceed 1.0.")
+
+            colors = FigureFactory._convert_to_RGB_255(colors)
+            colors = FigureFactory._label_rgb(colors)
+
+            colors_list = []
+            colors_list.append(colors)
+            colors = colors_list
+
+        elif isinstance(colors, list):
+            new_colormap = []
+            for color in colors:
+                if 'rgb' in color:
```

Comment:
This could be more 🐫 (DRY). Why don't you make `colors` a `list` right in the beginning? Does that make sense?